tests/unit_tests/db_engine_specs/test_presto.py

```diff
@@ -106,3 +107,35 @@ def test_get_schema_from_engine_params() -> None:
         )
         is None
     )
+
+@mock.patch("superset.db_engine_specs.presto.PrestoEngineSpec.latest_partition")
+@pytest.mark.parametrize(
+    ["column_type", "column_value", "expected_value"],
+    [
+        (types.DATE(), "2023-05-01", "DATE '2023-05-01'"),
+        (types.TIMESTAMP(), "2023-05-01", "TIMESTAMP '2023-05-01'")
+    ]
+)
+def test_where_latest_partition_date(
```

Comment:
I'll remove `_date` from the name here since I've parameterised it for multiple types.